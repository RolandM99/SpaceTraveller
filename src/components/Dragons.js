import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Dragon from './Dragon';
import { getDragonFromApi } from '../redux/reducers';

const Dragons = () => {
  const dispatch = useDispatch();
  const data = useSelector((state) => state.reducers);

  useEffect(() => {
    if (!data.dragons[0]) {
      dispatch(getDragonFromApi());
    }
  }, []);

  return (
    <div className="rockets">
      {data.dragons.map((dragon) => (
        <Dragon key={dragon.id} data={dragon} />
      ))}
    </div>
  );
};

export default Dragons;
