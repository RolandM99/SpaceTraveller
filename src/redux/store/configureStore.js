import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import rocketReducer from '../reducers';

const reducer = combineReducers({
  rocketReducer,
  // additional reducers could be added here
});

const store = createStore(
  reducer,
  applyMiddleware(logger, thunk),
);

export default store;
