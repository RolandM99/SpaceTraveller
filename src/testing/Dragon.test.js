import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store/configureStore';
import Dragons from '../components/Dragons';

describe('Dragon Component test', () => {
  it('Renders Dragon Row', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <Dragons />
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
