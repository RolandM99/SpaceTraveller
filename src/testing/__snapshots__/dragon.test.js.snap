// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragon Component test Renders Dragon Row 1`] = `
<div
  className="rockets"
/>
`;
